builder/package.centos-7

```diff
@@ -0,0 +1,46 @@
+if [[ ! -d /tmp/output/centos-7 ]]; then
+  mkdir -p /tmp/output/centos-7
+fi
+
+fpm \
+  -s dir \
+  -t rpm \
+  -v 1 \
+  -n R-${R_VERSION} \
+  --vendor "R Project" \
+  --deb-priority "optional" \
+  --deb-field 'Bugs: https://github.com/rstudio/r-builds/issues' \
+  --url "http://www.r-project.org/" \
+  --description "GNU R statistical computation and graphics system" \
+  --maintainer "RStudio Inc https://github.com/rstudio/r-builds" \
+  --license "GPL-2" \
+  -p /tmp/output/centos-7/ \
+  -d bzip2-devel \
+  -d gcc \
+  -d gcc-c++ \
+  -d gcc-gfortran \
+  -d libcurl-devel \
+  -d libicu-devel \
+  -d libSM \
+  -d libtiff \
+  -d libXmu \
+  -d libXt \
+  -d make \
+  -d openblas-devel \
```

Comment:
To use OpenBLAS with R on CentOS and openSUSE, there's an additional step of replacing the default BLAS with OpenBLAS. This is what we do in the CentOS 6/7 images: https://github.com/rstudio/r-docker/blob/c7e58d0be50259c7047a9f4eeb6da562ae3f7175/Dockerfile-centos.template#L16-L18

Can we do this automatically when users install the RPM, like as a post-install step?

Reply:
Ah, I didn't realise. We can add a post install step though, so I'll take a look at doing that.

Reply:
Post install script added

Reply:
Nice, the latest changes look good to me. I only noticed one minor issue while trying to uninstall. There were a bunch of empty directories left over, I think because the rpm doesn't know about `libRblas.so.keep`

Is there a way to make the rpm aware of that file? I don't see an obvious way since it's only created after installing.. If not, a workaround could be to create libRblas.so.keep before the packaging step.

```sh
[root@6911e123180b /]# yum -y remove R-3.5.3
Loaded plugins: fastestmirror, ovl
Resolving Dependencies
--> Running transaction check
---> Package R-3.5.3.x86_64 0:1-1 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================================================================================================================
 Package                                   Arch                                     Version                                  Repository                                   Size
===============================================================================================================================================================================
Removing:
 R-3.5.3                                   x86_64                                   1-1                                      installed                                    95 M

Transaction Summary
===============================================================================================================================================================================
Remove  1 Package

Installed size: 95 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Erasing    : R-3.5.3-1-1.x86_64                                                                                                                                          1/1 
  Verifying  : R-3.5.3-1-1.x86_64                                                                                                                                          1/1 

Removed:
  R-3.5.3.x86_64 0:1-1                                                                                                                                                         

Complete!

[root@6911e123180b /]# ls /opt/R/3.5.3/    
bin  lib  share

[root@6911e123180b /]# ls /opt/R/3.5.3/lib/R/lib     
libRblas.so.keep
```

Reply:
it might be cleanest to use a --before-remove script to remove it, or perhaps not keep it in the first place?

Reply:
Either of those sounds fine to me. I doubt most users would want to swap the default BLAS back in, but it'd still be nice to keep if there's an easy way to remove it.

Reply:
I created a 'before-remove' script to remove /opt/R/3.5.3/lib/R/lib/libRblas.so.keep, but the RPM system still knows that something in there has been changed and refuses to remove the /opt/R/3.5.3 directory tree.

In the end I've used an 'after-remove' script that deletes everything from /opt/R/R_VERSION down.

Reply:
Oh weird, maybe it's because we modified the original `libRblas.so` and would've had leftover directories anyway. This seems like the best solution then.